pkg/operator/controller/gateway-service-dns/controller.go

```diff
@@ -199,6 +199,11 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, nil
 	}
 
+	if !checkPlatformSupport(infraConfig) {
+		log.Info("infrastructure 'cluster' has unsupported configuration; reconciliation will be skipped")
+		return reconcile.Result{}, nil
+	}
+
 	domains := getGatewayHostnames(&gateway)
 	var errs []error
 	errs = append(errs, r.ensureDNSRecordsForGateway(ctx, &gateway, &service, domains.List(), infraConfig, dnsConfig)...)
```

Comment:
It can be useful to create the DNSRecord CR, but with `dnsManagementState: Unmanaged`, to serve as a reference for what DNS record the cluster-admin is _expected_ to create.  So I would suggest still calling `ensureDNSRecordsForGateway` but changing it to to set `dnsManagementState: Unmanaged` when the platform is not supported.  Does that make sense?

Reply:
> My reasoning for putting the check here vs `ensureDNSRecordsForGateway` was that this check would give the same outcome for each domain within `ensureDNSRecordsForGateway`.

If you're concerned about the performance impact, you can call `checkClusterHostedDNS` (previously `checkPlatformSupport`) in `Reconcile` and pass the Boolean result to `ensureDNSRecordsForGateway`.